connectors/sources/confluence.py

```diff
@@ -725,12 +843,19 @@ async def _attachment_coro(self, document, access_control):
 
     async def _space_coro(self):
         """Coroutine to add spaces documents to Queue"""
-        async for space, permissions in self.fetch_spaces():
-            access_control = list(
-                self._get_access_control_from_permission(
-                    permissions=permissions, target_type=SPACE
+        async for space, permissions, key in self.fetch_spaces():
+            if self.confluence_client.data_source_type == CONFLUENCE_CLOUD:
+                access_control = list(
+                    self._get_access_control_from_permission(
+                        permissions=permissions, target_type=SPACE
+                    )
+                )
+            else:
+                access_control = list(
+                    await self.fetch_server_space_permission(
+                        space_key=key
+                    )  # pyright: ignore
```

Comment:
I think it'll work if you define `access_control` outside of scope of if/else.

`# pyright: ignore` should be used only for cases when it's not solvable IMO. Can you fix it, instead of ignoring?